xlcalculator/xlfunctions/func_xltypes.py

```diff
@@ -267,8 +267,16 @@ def __datetime__(self):
     def __Blank__(self):
         return self.__class__('')
 
-    def upper(self):
-        return self.value.upper()
+    def __eq__(self, other):
+        if self.value in [None, ''] \
+                and (other in [None, ''] or isinstance(other, None)):
+            return True
+
+        else:
```

Comment:
I would not write else here, since the if returns. It makes it clear that the method always has a non-None result.